electron/src/lib/openGraph.ts

```diff
@@ -17,91 +17,100 @@
-  return new Promise(resolve => {
-    const imageRequest = request({encoding: null, url: encodeURI(url)}, (error, response, body: string) => {
-      if (!error && response.statusCode === 200) {
-        resolve(bufferToBase64(body, response.headers['content-type']));
-      } else {
-        // we just skip images that failed to download
-        resolve();
-      }
-    });
 
-    let currentDownloadSize = 0;
-    imageRequest.on('data', buffer => {
-      currentDownloadSize += buffer.length;
-      if (currentDownloadSize > IMAGE_SIZE_LIMIT) {
-        imageRequest.abort();
-        resolve();
-      }
-    });
-  });
-};
+  const axiosConfig: AxiosRequestConfig = {
+    headers: {
+      'User-Agent': USER_AGENT,
+    },
+    maxContentLength: IMAGE_SIZE_LIMIT,
+    method: 'get',
+    responseType: 'arraybuffer',
+    url,
+  };
+
+  let response;
+
+  try {
+    response = await axios.request<Buffer>(axiosConfig);
+  } catch (error) {
+    // we just skip images that failed to download
+    return;
+  }
 
-const fetchOpenGraphData = (url: string): Promise<OpenGraphResult> => {
-  const CONTENT_SIZE_LIMIT = 1e6; // ~1MB
-  const parsedUrl = urlUtil.parse(url);
-  const normalizedUrl = parsedUrl.protocol ? parsedUrl : urlUtil.parse(`http://${url}`);
+  if (response.status !== 200) {
+    // we just skip images that failed to download
+    return;
+  }
 
-  return new Promise<string>((resolve, reject) => {
-    let partialBody = '';
-    const getContentRequest = request.get({
-      headers: {
-        'User-Agent': USER_AGENT,
-      },
-      url: urlUtil.format(normalizedUrl),
-    });
+  const contentType = response.headers['content-type'] || '';
 
-    getContentRequest.on('response', response => {
-      if (response.statusCode !== 200) {
-        return reject(`Request failed with status code "${response.statusCode}".`);
-      }
+  return bufferToBase64(response.data, contentType);
+};
+
+const axiosContentLimit = (config: AxiosRequestConfig, contentLimit: number): Promise<string> => {
+  config.responseType = 'stream';
 
-      const contentType = response.headers['content-type'] || '';
-      const isHtmlContentType = contentType.match(/.*text\/html/);
+  return new Promise((resolve, reject) => {
+    let partialBody = '';
 
-      if (!isHtmlContentType) {
-        getContentRequest.abort();
-        throw new Error(`Unhandled format for open graph generation ('${contentType}')`);
+    return axios.request<NodeJS.ReadableStream>(config).then(response => {
+      if (response.status !== 200) {
+        return reject(`Request failed with status code "${response.status}".`);
       }
-    });
 
-    getContentRequest.on('data', buffer => {
-      const chunk = buffer.toString();
+      response.data.on('data', buffer => {
+        const chunk = buffer.toString();
+        partialBody += chunk;
 
-      partialBody += chunk;
+        if (chunk.match('</head>') || partialBody.length > contentLimit) {
+          resolve(partialBody);
```

Comment:
I think you should also abort the request. Else the browser will continue downloading content